src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs

```diff
@@ -760,7 +760,8 @@ await foreach (DocumentItem item in _source.GetDocumentsAsync(_options.Collectio
                         await compareExchangeActions.WriteKeyValueAsync(key, null, item.Document);
                         continue;
                     }
-
+                    if (compareExchangeActions?.MaybeFlush() == true) // we don't want to keep the cmp xng transaction open for too long
```

Comment:
This can be inside the previous condition:
`
if (compareExchangeActions != null)
{
    if (item.Document.ChangeVector != null && item.Document.ChangeVector.Contains(ChangeVectorParser.TrxnTag))
    {
        var key = ClusterTransactionCommand.GetAtomicGuardKey(item.Document.Id);
        await compareExchangeActions.WriteKeyValueAsync(key, null, item.Document);
        continue;
    }

    if (compareExchangeActions.MaybeFlush()) // we don't want to keep the cmp xng transaction open for too long
        await compareExchangeActions.FlushAsync();
}
`

Reply:
See the recent change